### DISCUSSION TITLE
Image input size for UNETR

### DISCUSSION THREAD

**Discussion by ouyuxuanbridge** [Project-MONAI/MONAI#3993] [Q&A]
Hi there,

firstly many thanks for your work.

I am very interested in UNETR.

I am wondering that if it can accept input of any size.

Thanks a lot

**Comment by ahatamiz** [comment]
Hi @Ouyuxuan623 

Thanks for your interest. As indicated in the [UNETR docs](https://docs.monai.io/en/stable/_modules/monai/networks/nets/unetr.html), any input size can be passed via `img_size`, provided that each dimension is divisible by 16, which is the patch size. For instance input sizes of `(96,96,96)` or `(32,32,32)` satisfy this requirement. 

Thanks

**Comment by fanqiehc** [comment]
May I use variable input size?
I want to train the net with (144, 144, 144), and (96, 96, 96) for inference.

**Comment by ahatamiz** [comment]
Hi @fanqiehc 

You can use a different resolution for inference, however the patch embedding layer need to be resampled back to the original training resolution before feeding into the network.

**Comment by muk465** [comment]
@ahatamiz sir does swinunetr also has this restriction ,of patch size being divisible by 16?
also can you please tell how to select these parameters for sliding window inference:

I have 256*256*256 mri volume i am using random cropped patches  of size 128*128*128 for training my swinunetr /unetr 
with batch size of 1  
how do i select 
1)roi size
2)sw_batch size
3)overlap between patches
during testing?

**Reply by ahatamiz** [reply]
Hi @muk465 an roi_size of [128,128,128] with sw_batchsize of 4 and overlap of 0.6 seem reasonable, if memory allows. Increasing overlap for inference to 0.7 maybe be useful.

**Reply by muk465** [reply]
@ahatamiz sir what i want to know is how you came up with these parameters ,like whats the intuition behind the sw_batch size and overlap?

**Reply by muk465** [reply]
because if memory doesnt allow i am think of decreasing patch size to 96*96*96 then how will you select sw_batch size and overlap ?

**Reply by ahatamiz** [reply]
Sw_batch size is the batch size of extracted patches. Sampling multiple patches from the same image is useful and 2 or 4 should work for this case ( based on previous experments).

Overlap is just the overlap of sliding window operation. Normally, larger number yields more accurate results. 

Sure, decreasing to 96 should also work, but again this is a hyper-paramer that needs to be validated.

**Reply by muk465** [reply]
Thanks for replying sir
